addons/popochiu/engine/objects/inventory_item/popochiu_inventory_item.gd

```diff
@@ -158,22 +207,34 @@ func get_description() -> String:
 
 
 # ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ PRIVATE ░░░░
-func _toggle_description(display: bool) -> void:
-	Cursor.set_cursor(cursor if display else CURSOR.Type.IDLE)
-	G.show_info(self.description if display else '')
-	if display:
-		description_toggled.emit(description if description else script_name)
+func _toggle_description(is_hover: bool) -> void:
+	if is_hover:
+		G.mouse_entered_inventory_item.emit(self)
 	else:
-		description_toggled.emit('')
+		last_click_button = -1
+		
+		G.mouse_exited_inventory_item.emit(self)
+	
+	# NOTE: Not sure this should go here
+	#Cursor.set_cursor(cursor if is_hover else CURSOR.Type.IDLE)
+	#G.show_hover_text(self.description if is_hover else '')
 
 
-func _on_action_pressed(event: InputEvent) -> void: 
+func _on_gui_input(event: InputEvent) -> void: 
 	var mouse_event := event as InputEventMouseButton 
-	if mouse_event:
-		if mouse_event.is_action_pressed('popochiu-interact'):
-			if I.active:
-				_on_item_used(I.active)
-			else:
-				_on_click()
-		elif mouse_event.is_action_pressed('popochiu-look'):
-			_on_right_click()
+	if mouse_event and mouse_event.pressed:
+		I.clicked = self
+		last_click_button = mouse_event.button_index
+		
+		match mouse_event.button_index:
+			MOUSE_BUTTON_LEFT:
+				if I.active:
+					_on_item_used(I.active)
+				else:
+					handle_command(mouse_event.button_index)
+			MOUSE_BUTTON_RIGHT:
```

Comment:
You may squeeze these in a single handler

Reply:
Done.